xml_converter/src/icon_gen.cpp

```diff
@@ -8,213 +10,456 @@ string Icon::classname() {
+        this->category_is_true = true;
     }
-    else if (attributename == "schedule") {
-        this->schedule = parse_string(attribute, errors);
+    else if (attributename == "category") {
+        this->category = parse_marker_category(attribute, errors);
+        this->category_is_true = true;
     }
     else if (attributename == "scheduleduration") {
         this->schedule_duration = parse_float(attribute, errors);
+        this->schedule_duration_is_true = true;
     }
-    else if (attributename == "show") {
-        this->show_category = parse_marker_category(attribute, errors);
+    else if (attributename == "schedule") {
+        this->schedule = parse_string(attribute, errors);
+        this->schedule_is_true = true;
     }
-    else if (attributename == "specialization") {
-        this->specialization_filter = parse_specialization_filter(attribute, errors);
+    else if (attributename == "color") {
+        this->color = parse_color(attribute, errors);
+        this->color_is_true = true;
     }
-    else if (attributename == "race") {
-        this->species_filter = parse_species_filter(attribute, errors);
+    else if (attributename == "iconfile") {
+        this->icon = parse_image(attribute, errors);
+        this->icon_is_true = true;
     }
-    else if (attributename == "species") {
-        this->species_filter = parse_species_filter(attribute, errors);
+    else if (attributename == "canfade") {
+        this->can_fade = parse_bool(attribute, errors);
+        this->can_fade_is_true = true;
     }
-    else if (attributename == "toggle") {
-        this->toggle_category = parse_marker_category(attribute, errors);
+    else if (attributename == "rotatex") {
+        this->x_rotation = parse_float(attribute, errors);
+        this->x_rotation_is_true = true;
     }
-    else if (attributename == "tipname") {
-        this->tooltip_description = parse_string(attribute, errors);
+    else if (attributename == "rotatey") {
+        this->y_rotation = parse_float(attribute, errors);
+        this->y_rotation_is_true = true;
     }
-    else if (attributename == "tipdescription") {
-        this->tooltip_name = parse_string(attribute, errors);
+    else if (attributename == "rotatez") {
+        this->z_rotation = parse_float(attribute, errors);
+        this->z_rotation_is_true = true;
     }
-    else if (attributename == "triggerrange") {
-        this->trigger_range = parse_float(attribute, errors);
+    else if (attributename == "rotate") {
+        this->z_rotation = parse_float(attribute, errors);
+        this->z_rotation_is_true = true;
     }
-    else if (attributename == "inforange") {
-        this->trigger_range = parse_float(attribute, errors);
+    else if (attributename == "rotatez") {
+        this->euler_rotation = parse_euler_rotation(attribute, errors);
+        this->euler_rotation_is_true = true;
+    }
+    else if (attributename == "rotate") {
```

Comment:
You have `rotate` multiple times here

Reply:
Found the logic error. Components now have a different variable name in the loop so the value cannot escape.